Enable sklearnex documentation (#587)
intel/scikit-learn-intelex: doc/sources/index.rst

Usage
--------------------
Intel(R) Extension for Scikit-learn* dynamically patches scikit-learn estimators to use Intel(R) oneAPI Data Analytics Library
as the underlying solver, while getting the same solution faster.

- It is possible to enable those patches without editing the code of a scikit-learn application by
  using the following commandline flag::

    python -m sklearnex my_application.py

- Or from your script::

    from sklearnex import patch_sklearn
    patch_sklearn()

For example::

    from sklearnex import patch_sklearn
    patch_sklearn()

    from sklearn.cluster import KMeans
    import numpy as np
    X = np.array([[1, 2], [1, 4], [1, 0],
                  [10, 2], [10, 4], [10, 0]])
    kmeans = KMeans(n_clusters=2, random_state=0).fit(X)
    print(f"kmeans.labels_ = {kmeans.labels_}")
    pred = kmeans.predict([[0, 0], [12, 3]])
    print(f"pred = {pred}")
    print(f"kmeans.cluster_centers_ = {kmeans.cluster_centers_}")

In the example above, you can see that the use of the original Scikit-learn
has not changed. This behavior is achieved through drop-in patching.

To undo the patch Scikit-learn with::

    sklearnex.unpatch_sklearn()

> Intel (R) Extension for Scikit-learn * does not patch all scikit-learn algorithms and parameters.

Reviewer: ```suggestion
Intel (R) Extension for Scikit-learn* does not patch all scikit-learn algorithms and parameters.
```

You can find the :ref:`full patching map here <sklearn_algorithms>`.

.. note::
    Intel(R) Extension for Scikit-learn* supports optimizations for the last four versions of scikit-learn.

>     The latest release of scikit-learn-intelex-2021.2.1 supports scikit-learn 0.21.X, 0.22.X, 0.23.X and 0.24.X.

Reviewer 2: ```suggestion
    The latest release of scikit-learn-intelex-2021.2.X supports scikit-learn 0.21.X, 0.22.X, 0.23.X and 0.24.X.
```

Follow us on Medium
--------------------
We publish blogs on Medium, so `follow us <https://medium.com/intel-analytics-software/tagged/machine-learning>`_
to learn tips and tricks for more efficient data analysis the help of Intel(R) Extension for Scikit-learn.
Here are our latest blogs:

> 

Reviewer 2: ```suggestion

- [Intel Gives Scikit-Learn the Performance Boost Data Scientists Need](https://medium.com/intel-analytics-software/intel-gives-scikit-learn-the-performance-boost-data-scientists-need-42eb47c80b18)
```

.. _index:

#####################################################
Intel(R) Extension for Scikit-learn*
#####################################################

> Intel(R) Extension for Scikit-learn speeds up scikit-learn beyond by providing drop-in patching.

Reviewer 3: ```suggestion
Intel(R) Extension for Scikit-learn speeds up scikit-learn by providing drop-in patching.
```

Acceleration is achieved through the use of the Intel(R) oneAPI Data Analytics Library (oneDAL)
that allows for fast usage of the framework suited for Data Scientists or Machine Learning users.

Designed for Data Scientists and Framework Designers
----------------------------------------------------

> Intel(R) Extension for Scikit-learn* was created to give data scientists the easiest way to get performance, 
> when using familiar Scikit-learn package.

Reviewer 3: ```suggestion
Intel(R) Extension for Scikit-learn* was created to give data scientists the easiest way to get a better performance
while using the familiar scikit-learn package.
```

Usage
--------------------
Intel(R) Extension for Scikit-learn* dynamically patches scikit-learn estimators to use Intel(R) oneAPI Data Analytics Library
as the underlying solver, while getting the same solution faster.

- It is possible to enable those patches without editing the code of a scikit-learn application by
  using the following commandline flag::

    python -m sklearnex my_application.py

- Or from your script::

    from sklearnex import patch_sklearn
    patch_sklearn()

For example::

    from sklearnex import patch_sklearn
    patch_sklearn()

    from sklearn.cluster import KMeans
    import numpy as np
    X = np.array([[1, 2], [1, 4], [1, 0],
                  [10, 2], [10, 4], [10, 0]])
    kmeans = KMeans(n_clusters=2, random_state=0).fit(X)
    print(f"kmeans.labels_ = {kmeans.labels_}")
    pred = kmeans.predict([[0, 0], [12, 3]])
    print(f"pred = {pred}")

>     print(f"kmeans.cluster_centers_ = {kmeans.cluster_centers_}")

Reviewer 3: could you please check that this is rendered in the output with the proper syntax highlighting?

Author: Yes, syntax is highlighting correctly
![image](https://user-images.githubusercontent.com/35104773/112964619-2b506680-9151-11eb-9550-cb03b408ea96.png)
